In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
get_ipython().run_line_magic('matplotlib', 'inline')
import seaborn as sns


In [2]:
#messy dataset_movies.csv

In [3]:
df=pd.read_csv("messy dataset_movies.csv")
df.head(10)

MOVIES         YEAR  \
0                        Blood Red Sky       (2021)   
1  Masters of the Universe: Revelation     (2021– )   
2                     The Walking Dead  (2010–2022)   
3                       Rick and Morty     (2013– )   
4                      Army of Thieves       (2021)   
5                          Outer Banks     (2020– )   
6      The Last Letter from Your Lover       (2021)   
7                               Dexter  (2006–2013)   
8                    Never Have I Ever     (2020– )   
9                         Virgin River     (2019– )   

                                        GENRE  RATING  \
0      \nAction, Horror, Thriller                 6.1   
1  \nAnimation, Action, Adventure                 5.0   
2       \nDrama, Horror, Thriller                 8.2   
3  \nAnimation, Adventure, Comedy                 9.2   
4         \nAction, Crime, Horror                 NaN   
5          \nAction, Crime, Drama                 7.6   
6                \nDrama, Romance                 6.8   
7         \nCrime, Drama, Mystery                 8.6   
8                        \nComedy                 7.9   
9                \nDrama, Romance                 7.4   

                                            ONE-LINE  \
0  \nA woman with a mysterious illness is forced ...   
1  \nThe war for Eternia begins again in what may...   
2  \nSheriff Deputy Rick Grimes wakes up from a c...   
3  \nAn animated series that follows the exploits...   
4  \nA prequel, set before the events of Army of ...   
5  \nA group of teenagers from the wrong side of ...   
6  \nA pair of interwoven stories set in the past...   
7  \nBy day, mild-mannered Dexter is a blood-spat...   
8  \nThe complicated life of a modern-day first g...   
9  \nSeeking a fresh start, nurse practitioner Me...   

                                               STARS    VOTES  RunTime Gross  
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   NaN  
1  \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   NaN  
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   NaN  
3  \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   NaN  
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   NaN  
5  \n            \n    Stars:\nChase Stokes, \nMa...   25,858     50.0   NaN  
6  \n    Director:\nAugustine Frizzell\n| \n    S...    5,283    110.0   NaN  
7  \n            \n    Stars:\nMichael C. Hall, \...  665,387     53.0   NaN  
8  \n            \n    Stars:\nMaitreyi Ramakrish...   34,530     30.0   NaN  
9  \n            \n    Stars:\nAlexandra Breckenr...   27,279     44.0   NaN

# #Renameing the columns

In [4]:
df=df.rename(columns={'RunTime':'RUNTIME'})
df.head()

MOVIES         YEAR  \
0                        Blood Red Sky       (2021)   
1  Masters of the Universe: Revelation     (2021– )   
2                     The Walking Dead  (2010–2022)   
3                       Rick and Morty     (2013– )   
4                      Army of Thieves       (2021)   

                                        GENRE  RATING  \
0      \nAction, Horror, Thriller                 6.1   
1  \nAnimation, Action, Adventure                 5.0   
2       \nDrama, Horror, Thriller                 8.2   
3  \nAnimation, Adventure, Comedy                 9.2   
4         \nAction, Crime, Horror                 NaN   

                                            ONE-LINE  \
0  \nA woman with a mysterious illness is forced ...   
1  \nThe war for Eternia begins again in what may...   
2  \nSheriff Deputy Rick Grimes wakes up from a c...   
3  \nAn animated series that follows the exploits...   
4  \nA prequel, set before the events of Army of ...   

                                               STARS    VOTES  RUNTIME Gross  
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   NaN  
1  \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   NaN  
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   NaN  
3  \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   NaN  
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   NaN

# finding missing values

In [5]:
df.isnull().any()

MOVIES      False
YEAR         True
GENRE        True
RATING       True
ONE-LINE    False
STARS       False
VOTES        True
RUNTIME      True
Gross        True
dtype: bool

In [6]:
df.isnull().sum()

MOVIES         0
YEAR         644
GENRE         80
RATING      1820
ONE-LINE       0
STARS          0
VOTES       1820
RUNTIME     2958
Gross       9539
dtype: int64

In [7]:
df.shape

(9999, 9)

In [8]:
df.columns

Index(['MOVIES', 'YEAR', 'GENRE', 'RATING', 'ONE-LINE', 'STARS', 'VOTES',
       'RUNTIME', 'Gross'],
      dtype='object')

# we should delete the whole column becouse column Gross has 9539 null values 

In [9]:
del df['Gross']

In [10]:
df.columns

Index(['MOVIES', 'YEAR', 'GENRE', 'RATING', 'ONE-LINE', 'STARS', 'VOTES',
       'RUNTIME'],
      dtype='object')

In [11]:
df['VOTES'].head(50)
#df.info()

0        21,062
1        17,870
2       885,805
3       414,849
4           NaN
5        25,858
6         5,283
7       665,387
8        34,530
9        27,279
10       17,989
11      264,222
12       50,148
13       12,172
14      286,488
15      266,258
16    1,552,311
17       70,871
18       71,544
19      885,856
20       36,634
21      880,918
22      320,264
23      393,414
24          NaN
25        4,681
26      459,913
27      347,857
28      404,273
29      213,522
30       27,447
31       26,926
32      248,583
33      125,619
34       34,906
35      378,030
36       73,656
37      336,484
38      171,804
39      242,582
40      374,263
41      341,858
42       90,728
43      134,991
44       94,650
45      469,407
46      206,090
47      169,517
48       37,142
49      132,378
Name: VOTES, dtype: object

# we replace the null values of RUNTIME of the movies as mean of the series/column

In [12]:
df['RUNTIME'].fillna(df['RUNTIME'].mean(),inplace=True)
df.isnull().sum()

MOVIES         0
YEAR         644
GENRE         80
RATING      1820
ONE-LINE       0
STARS          0
VOTES       1820
RUNTIME        0
dtype: int64

# here we replace the comma in the integer column and convert it into the float

In [13]:
df['VOTES'].replace(',','', regex=True, inplace=True)
df['VOTES']=df['VOTES'].astype(str).astype(float)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9999 entries, 0 to 9998
Data columns (total 8 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   MOVIES    9999 non-null   object 
 1   YEAR      9355 non-null   object 
 2   GENRE     9919 non-null   object 
 3   RATING    8179 non-null   float64
 4   ONE-LINE  9999 non-null   object 
 5   STARS     9999 non-null   object 
 6   VOTES     8179 non-null   float64
 7   RUNTIME   9999 non-null   float64
dtypes: float64(3), object(5)
memory usage: 625.1+ KB


In [14]:
df['VOTES'].head(50)

0       21062.0
1       17870.0
2      885805.0
3      414849.0
4           NaN
5       25858.0
6        5283.0
7      665387.0
8       34530.0
9       27279.0
10      17989.0
11     264222.0
12      50148.0
13      12172.0
14     286488.0
15     266258.0
16    1552311.0
17      70871.0
18      71544.0
19     885856.0
20      36634.0
21     880918.0
22     320264.0
23     393414.0
24          NaN
25       4681.0
26     459913.0
27     347857.0
28     404273.0
29     213522.0
30      27447.0
31      26926.0
32     248583.0
33     125619.0
34      34906.0
35     378030.0
36      73656.0
37     336484.0
38     171804.0
39     242582.0
40     374263.0
41     341858.0
42      90728.0
43     134991.0
44      94650.0
45     469407.0
46     206090.0
47     169517.0
48      37142.0
49     132378.0
Name: VOTES, dtype: float64

In [15]:
df['VOTES'].fillna(df['VOTES'].mean() ,inplace=True)
df.isnull().sum()

MOVIES         0
YEAR         644
GENRE         80
RATING      1820
ONE-LINE       0
STARS          0
VOTES          0
RUNTIME        0
dtype: int64

In [16]:
df['RATING'].fillna(df['RATING'].mean() ,inplace=True)
df.isnull().sum()

MOVIES        0
YEAR        644
GENRE        80
RATING        0
ONE-LINE      0
STARS         0
VOTES         0
RUNTIME       0
dtype: int64

In [17]:
df.dropna(inplace=True)
df.isnull().sum()

MOVIES      0
YEAR        0
GENRE       0
RATING      0
ONE-LINE    0
STARS       0
VOTES       0
RUNTIME     0
dtype: int64

# now the null values handled so finding the duplicates

In [18]:
df.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
9994     True
9995    False
9996    False
9997    False
9998    False
Length: 9316, dtype: bool

In [19]:
df[df.duplicated()==1]

MOVIES      YEAR  \
6833   Mighty Little Bheem  (2019– )   
6835   Mighty Little Bheem  (2019– )   
6836   Mighty Little Bheem  (2019– )   
6837   Mighty Little Bheem  (2019– )   
6838   Mighty Little Bheem  (2019– )   
...                    ...       ...   
9989                  1899  (2022– )   
9990                  1899  (2022– )   
9991                  1899  (2022– )   
9992                  1899  (2022– )   
9994        The Imperfects  (2021– )   

                                          GENRE    RATING        ONE-LINE  \
6833  \nAnimation, Short, Adventure              6.921176  \nAdd a Plot\n   
6835  \nAnimation, Short, Adventure              9.000000  \nAdd a Plot\n   
6836  \nAnimation, Short, Adventure              9.000000  \nAdd a Plot\n   
6837  \nAnimation, Short, Adventure              6.921176  \nAdd a Plot\n   
6838  \nAnimation, Short, Adventure              6.921176  \nAdd a Plot\n   
...                                         ...       ...             ...   
9989       \nDrama, History, Horror              6.921176  \nAdd a Plot\n   
9990       \nDrama, History, Horror              6.921176  \nAdd a Plot\n   
9991       \nDrama, History, Horror              6.921176  \nAdd a Plot\n   
9992       \nDrama, History, Horror              6.921176  \nAdd a Plot\n   
9994    \nAdventure, Drama, Fantasy              6.921176  \nAdd a Plot\n   

                                                  STARS         VOTES  \
6833  \n    Directors:\nRajiv Chilaka, \nKrishna Moh...  15124.062722   
6835  \n    Directors:\nRajiv Chilaka, \nKrishna Moh...      6.000000   
6836  \n    Directors:\nRajiv Chilaka, \nKrishna Moh...      6.000000   
6837  \n    Directors:\nRajiv Chilaka, \nKrishna Moh...  15124.062722   
6838  \n    Directors:\nRajiv Chilaka, \nKrishna Moh...  15124.062722   
...                                                 ...           ...   
9989                   \n    Director:\nBaran bo Odar\n  15124.062722   
9990                   \n    Director:\nBaran bo Odar\n  15124.062722   
9991                   \n    Director:\nBaran bo Odar\n  15124.062722   
9992                   \n    Director:\nBaran bo Odar\n  15124.062722   
9994  \n            \n    Stars:\nMorgan Taylor Camp...  15124.062722   

        RUNTIME  
6833  68.688539  
6835  68.688539  
6836  68.688539  
6837  68.688539  
6838  68.688539  
...         ...  
9989  68.688539  
9990  68.688539  
9991  68.688539  
9992  68.688539  
9994  68.688539  

[327 rows x 8 columns]

# we have 327 duplicates

In [20]:
df=df.drop_duplicates()

In [21]:
df.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
9993    False
9995    False
9996    False
9997    False
9998    False
Length: 8989, dtype: bool

In [22]:
df.duplicated().sum()

0

In [23]:
df.head()

MOVIES         YEAR  \
0                        Blood Red Sky       (2021)   
1  Masters of the Universe: Revelation     (2021– )   
2                     The Walking Dead  (2010–2022)   
3                       Rick and Morty     (2013– )   
4                      Army of Thieves       (2021)   

                                        GENRE    RATING  \
0      \nAction, Horror, Thriller              6.100000   
1  \nAnimation, Action, Adventure              5.000000   
2       \nDrama, Horror, Thriller              8.200000   
3  \nAnimation, Adventure, Comedy              9.200000   
4         \nAction, Crime, Horror              6.921176   

                                            ONE-LINE  \
0  \nA woman with a mysterious illness is forced ...   
1  \nThe war for Eternia begins again in what may...   
2  \nSheriff Deputy Rick Grimes wakes up from a c...   
3  \nAn animated series that follows the exploits...   
4  \nA prequel, set before the events of Army of ...   

                                               STARS          VOTES  \
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21062.000000   
1  \n            \n    Stars:\nChris Wood, \nSara...   17870.000000   
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885805.000000   
3  \n            \n    Stars:\nJustin Roiland, \n...  414849.000000   
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...   15124.062722   

      RUNTIME  
0  121.000000  
1   25.000000  
2   44.000000  
3   23.000000  
4   68.688539

In [24]:
df['GENRE'].values

array(['\nAction, Horror, Thriller            ',
       '\nAnimation, Action, Adventure            ',
       '\nDrama, Horror, Thriller            ', ...,
       '\nDocumentary, Sport            ',
       '\nAdventure, Drama, Fantasy            ',
       '\nAdventure, Drama, Fantasy            '], dtype=object)

# in GENRE column we have messy data as \n and comma

In [25]:
df['GENRE']=df['GENRE'].str.split(pat='\n').str[1]
df.head()

MOVIES         YEAR  \
0                        Blood Red Sky       (2021)   
1  Masters of the Universe: Revelation     (2021– )   
2                     The Walking Dead  (2010–2022)   
3                       Rick and Morty     (2013– )   
4                      Army of Thieves       (2021)   

                                      GENRE    RATING  \
0      Action, Horror, Thriller              6.100000   
1  Animation, Action, Adventure              5.000000   
2       Drama, Horror, Thriller              8.200000   
3  Animation, Adventure, Comedy              9.200000   
4         Action, Crime, Horror              6.921176   

                                            ONE-LINE  \
0  \nA woman with a mysterious illness is forced ...   
1  \nThe war for Eternia begins again in what may...   
2  \nSheriff Deputy Rick Grimes wakes up from a c...   
3  \nAn animated series that follows the exploits...   
4  \nA prequel, set before the events of Army of ...   

                                               STARS          VOTES  \
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21062.000000   
1  \n            \n    Stars:\nChris Wood, \nSara...   17870.000000   
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885805.000000   
3  \n            \n    Stars:\nJustin Roiland, \n...  414849.000000   
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...   15124.062722   

      RUNTIME  
0  121.000000  
1   25.000000  
2   44.000000  
3   23.000000  
4   68.688539

In [26]:
df['GENRE1']=df['GENRE'].str.split(pat=',').str[0]
df['GENRE2']=df['GENRE'].str.split(pat=',').str[1]
df['GENRE3']=df['GENRE'].str.split(pat=',').str[2]
df.head()

MOVIES         YEAR  \
0                        Blood Red Sky       (2021)   
1  Masters of the Universe: Revelation     (2021– )   
2                     The Walking Dead  (2010–2022)   
3                       Rick and Morty     (2013– )   
4                      Army of Thieves       (2021)   

                                      GENRE    RATING  \
0      Action, Horror, Thriller              6.100000   
1  Animation, Action, Adventure              5.000000   
2       Drama, Horror, Thriller              8.200000   
3  Animation, Adventure, Comedy              9.200000   
4         Action, Crime, Horror              6.921176   

                                            ONE-LINE  \
0  \nA woman with a mysterious illness is forced ...   
1  \nThe war for Eternia begins again in what may...   
2  \nSheriff Deputy Rick Grimes wakes up from a c...   
3  \nAn animated series that follows the exploits...   
4  \nA prequel, set before the events of Army of ...   

                                               STARS          VOTES  \
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21062.000000   
1  \n            \n    Stars:\nChris Wood, \nSara...   17870.000000   
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885805.000000   
3  \n            \n    Stars:\nJustin Roiland, \n...  414849.000000   
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...   15124.062722   

      RUNTIME     GENRE1      GENRE2                  GENRE3  
0  121.000000     Action      Horror    Thriller              
1   25.000000  Animation      Action   Adventure              
2   44.000000      Drama      Horror    Thriller              
3   23.000000  Animation   Adventure      Comedy              
4   68.688539     Action       Crime      Horror

In [27]:
df['Director']=df['STARS'].str.split(pat='Director:').str[1]


#df['STARS3']=df['STARS'].str.split(pat='\n').str[3]
#f['STARS4']=df['STARS'].str.split(pat='\n').str[4]
df.head()

MOVIES         YEAR  \
0                        Blood Red Sky       (2021)   
1  Masters of the Universe: Revelation     (2021– )   
2                     The Walking Dead  (2010–2022)   
3                       Rick and Morty     (2013– )   
4                      Army of Thieves       (2021)   

                                      GENRE    RATING  \
0      Action, Horror, Thriller              6.100000   
1  Animation, Action, Adventure              5.000000   
2       Drama, Horror, Thriller              8.200000   
3  Animation, Adventure, Comedy              9.200000   
4         Action, Crime, Horror              6.921176   

                                            ONE-LINE  \
0  \nA woman with a mysterious illness is forced ...   
1  \nThe war for Eternia begins again in what may...   
2  \nSheriff Deputy Rick Grimes wakes up from a c...   
3  \nAn animated series that follows the exploits...   
4  \nA prequel, set before the events of Army of ...   

                                               STARS          VOTES  \
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21062.000000   
1  \n            \n    Stars:\nChris Wood, \nSara...   17870.000000   
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885805.000000   
3  \n            \n    Stars:\nJustin Roiland, \n...  414849.000000   
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...   15124.062722   

      RUNTIME     GENRE1      GENRE2                  GENRE3  \
0  121.000000     Action      Horror    Thriller               
1   25.000000  Animation      Action   Adventure               
2   44.000000      Drama      Horror    Thriller               
3   23.000000  Animation   Adventure      Comedy               
4   68.688539     Action       Crime      Horror               

                                            Director  
0  \nPeter Thorwarth\n| \n    Stars:\nPeri Baumei...  
1                                                NaN  
2                                                NaN  
3                                                NaN  
4  \nMatthias Schweighöfer\n| \n    Stars:\nMatth...

In [28]:
df['STARS1']=df['Director'].str.split(pat='Stars:').str[1]

In [29]:
df.head(1)

MOVIES    YEAR                                 GENRE  RATING  \
0  Blood Red Sky  (2021)  Action, Horror, Thriller                 6.1   

                                            ONE-LINE  \
0  \nA woman with a mysterious illness is forced ...   

                                               STARS    VOTES  RUNTIME  \
0  \n    Director:\nPeter Thorwarth\n| \n    Star...  21062.0    121.0   

   GENRE1   GENRE2                 GENRE3  \
0  Action   Horror   Thriller               

                                            Director  \
0  \nPeter Thorwarth\n| \n    Stars:\nPeri Baumei...   

                                              STARS1  
0  \nPeri Baumeister, \nCarl Anton Koch, \nAlexan...

In [30]:
df['STARS2']=df['Director'].str.split(pat='Stars:').str[2]
df.head(1)

MOVIES    YEAR                                 GENRE  RATING  \
0  Blood Red Sky  (2021)  Action, Horror, Thriller                 6.1   

                                            ONE-LINE  \
0  \nA woman with a mysterious illness is forced ...   

                                               STARS    VOTES  RUNTIME  \
0  \n    Director:\nPeter Thorwarth\n| \n    Star...  21062.0    121.0   

   GENRE1   GENRE2                 GENRE3  \
0  Action   Horror   Thriller               

                                            Director  \
0  \nPeter Thorwarth\n| \n    Stars:\nPeri Baumei...   

                                              STARS1  STARS2  
0  \nPeri Baumeister, \nCarl Anton Koch, \nAlexan...     NaN

In [31]:
df.isnull().sum()

MOVIES         0
YEAR           0
GENRE          0
RATING         0
ONE-LINE       0
STARS          0
VOTES          0
RUNTIME        0
GENRE1         0
GENRE2      2349
GENRE3      4393
Director    3941
STARS1      4376
STARS2      8989
dtype: int64

# so by spliting columns we found again some null values,we will not drop them just replace them

In [32]:
df['Director']=df['Director'].str.split(pat='\n').str[1]
df['Director']

0             Peter Thorwarth
1                         NaN
2                         NaN
3                         NaN
4       Matthias Schweighöfer
                ...          
9993            Nicolai Rohde
9995                      NaN
9996    Orlando von Einsiedel
9997         Jovanka Vuckovic
9998         Jovanka Vuckovic
Name: Director, Length: 8989, dtype: object

In [33]:
df['STARS1']=df['STARS1'].str.split(pat='\n').str[1]
df['STARS1']

0              Peri Baumeister, 
1                            NaN
2                            NaN
3                            NaN
4        Matthias Schweighöfer, 
                  ...           
9993               Felix Klare, 
9995                         NaN
9996                         NaN
9997    Morgan Taylor Campbell, 
9998    Morgan Taylor Campbell, 
Name: STARS1, Length: 8989, dtype: object

In [34]:
df.head(2)

MOVIES      YEAR  \
0                        Blood Red Sky    (2021)   
1  Masters of the Universe: Revelation  (2021– )   

                                      GENRE  RATING  \
0      Action, Horror, Thriller                 6.1   
1  Animation, Action, Adventure                 5.0   

                                            ONE-LINE  \
0  \nA woman with a mysterious illness is forced ...   
1  \nThe war for Eternia begins again in what may...   

                                               STARS    VOTES  RUNTIME  \
0  \n    Director:\nPeter Thorwarth\n| \n    Star...  21062.0    121.0   
1  \n            \n    Stars:\nChris Wood, \nSara...  17870.0     25.0   

      GENRE1   GENRE2                  GENRE3         Director  \
0     Action   Horror    Thriller              Peter Thorwarth   
1  Animation   Action   Adventure                          NaN   

              STARS1  STARS2  
0  Peri Baumeister,      NaN  
1                NaN     NaN

In [35]:
df['ONE-LINE']=df['ONE-LINE'].str.split(pat='\n').str[1]
df['ONE-LINE']

0       A woman with a mysterious illness is forced in...
1       The war for Eternia begins again in what may b...
2       Sheriff Deputy Rick Grimes wakes up from a com...
3       An animated series that follows the exploits o...
4       A prequel, set before the events of Army of th...
                              ...                        
9993                                           Add a Plot
9995                                           Add a Plot
9996                                           Add a Plot
9997                                           Add a Plot
9998                                           Add a Plot
Name: ONE-LINE, Length: 8989, dtype: object

In [36]:
df.head()

MOVIES         YEAR  \
0                        Blood Red Sky       (2021)   
1  Masters of the Universe: Revelation     (2021– )   
2                     The Walking Dead  (2010–2022)   
3                       Rick and Morty     (2013– )   
4                      Army of Thieves       (2021)   

                                      GENRE    RATING  \
0      Action, Horror, Thriller              6.100000   
1  Animation, Action, Adventure              5.000000   
2       Drama, Horror, Thriller              8.200000   
3  Animation, Adventure, Comedy              9.200000   
4         Action, Crime, Horror              6.921176   

                                            ONE-LINE  \
0  A woman with a mysterious illness is forced in...   
1  The war for Eternia begins again in what may b...   
2  Sheriff Deputy Rick Grimes wakes up from a com...   
3  An animated series that follows the exploits o...   
4  A prequel, set before the events of Army of th...   

                                               STARS          VOTES  \
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21062.000000   
1  \n            \n    Stars:\nChris Wood, \nSara...   17870.000000   
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885805.000000   
3  \n            \n    Stars:\nJustin Roiland, \n...  414849.000000   
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...   15124.062722   

      RUNTIME     GENRE1      GENRE2                  GENRE3  \
0  121.000000     Action      Horror    Thriller               
1   25.000000  Animation      Action   Adventure               
2   44.000000      Drama      Horror    Thriller               
3   23.000000  Animation   Adventure      Comedy               
4   68.688539     Action       Crime      Horror               

                Director                   STARS1  STARS2  
0        Peter Thorwarth        Peri Baumeister,      NaN  
1                    NaN                      NaN     NaN  
2                    NaN                      NaN     NaN  
3                    NaN                      NaN     NaN  
4  Matthias Schweighöfer  Matthias Schweighöfer,      NaN

# here we delete the old GENRE messy column

In [37]:
del df['GENRE']
df.head()

MOVIES         YEAR    RATING  \
0                        Blood Red Sky       (2021)  6.100000   
1  Masters of the Universe: Revelation     (2021– )  5.000000   
2                     The Walking Dead  (2010–2022)  8.200000   
3                       Rick and Morty     (2013– )  9.200000   
4                      Army of Thieves       (2021)  6.921176   

                                            ONE-LINE  \
0  A woman with a mysterious illness is forced in...   
1  The war for Eternia begins again in what may b...   
2  Sheriff Deputy Rick Grimes wakes up from a com...   
3  An animated series that follows the exploits o...   
4  A prequel, set before the events of Army of th...   

                                               STARS          VOTES  \
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21062.000000   
1  \n            \n    Stars:\nChris Wood, \nSara...   17870.000000   
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885805.000000   
3  \n            \n    Stars:\nJustin Roiland, \n...  414849.000000   
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...   15124.062722   

      RUNTIME     GENRE1      GENRE2                  GENRE3  \
0  121.000000     Action      Horror    Thriller               
1   25.000000  Animation      Action   Adventure               
2   44.000000      Drama      Horror    Thriller               
3   23.000000  Animation   Adventure      Comedy               
4   68.688539     Action       Crime      Horror               

                Director                   STARS1  STARS2  
0        Peter Thorwarth        Peri Baumeister,      NaN  
1                    NaN                      NaN     NaN  
2                    NaN                      NaN     NaN  
3                    NaN                      NaN     NaN  
4  Matthias Schweighöfer  Matthias Schweighöfer,      NaN

In [38]:
df['Director'].fillna("not specified",inplace = True)
df['STARS1'].fillna("not sure",inplace = True)
df['STARS2'].fillna("..",inplace = True)

df.head()

MOVIES         YEAR    RATING  \
0                        Blood Red Sky       (2021)  6.100000   
1  Masters of the Universe: Revelation     (2021– )  5.000000   
2                     The Walking Dead  (2010–2022)  8.200000   
3                       Rick and Morty     (2013– )  9.200000   
4                      Army of Thieves       (2021)  6.921176   

                                            ONE-LINE  \
0  A woman with a mysterious illness is forced in...   
1  The war for Eternia begins again in what may b...   
2  Sheriff Deputy Rick Grimes wakes up from a com...   
3  An animated series that follows the exploits o...   
4  A prequel, set before the events of Army of th...   

                                               STARS          VOTES  \
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21062.000000   
1  \n            \n    Stars:\nChris Wood, \nSara...   17870.000000   
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885805.000000   
3  \n            \n    Stars:\nJustin Roiland, \n...  414849.000000   
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...   15124.062722   

      RUNTIME     GENRE1      GENRE2                  GENRE3  \
0  121.000000     Action      Horror    Thriller               
1   25.000000  Animation      Action   Adventure               
2   44.000000      Drama      Horror    Thriller               
3   23.000000  Animation   Adventure      Comedy               
4   68.688539     Action       Crime      Horror               

                Director                   STARS1 STARS2  
0        Peter Thorwarth        Peri Baumeister,      ..  
1          not specified                 not sure     ..  
2          not specified                 not sure     ..  
3          not specified                 not sure     ..  
4  Matthias Schweighöfer  Matthias Schweighöfer,      ..

In [39]:
df.isnull().sum()

MOVIES         0
YEAR           0
RATING         0
ONE-LINE       0
STARS          0
VOTES          0
RUNTIME        0
GENRE1         0
GENRE2      2349
GENRE3      4393
Director       0
STARS1         0
STARS2         0
dtype: int64

In [40]:
df['GENRE1'].fillna("not sure",inplace=True)
df['GENRE2'].fillna("..",inplace=True)
df['GENRE3'].fillna("..",inplace=True)

In [41]:
df.head()

MOVIES         YEAR    RATING  \
0                        Blood Red Sky       (2021)  6.100000   
1  Masters of the Universe: Revelation     (2021– )  5.000000   
2                     The Walking Dead  (2010–2022)  8.200000   
3                       Rick and Morty     (2013– )  9.200000   
4                      Army of Thieves       (2021)  6.921176   

                                            ONE-LINE  \
0  A woman with a mysterious illness is forced in...   
1  The war for Eternia begins again in what may b...   
2  Sheriff Deputy Rick Grimes wakes up from a com...   
3  An animated series that follows the exploits o...   
4  A prequel, set before the events of Army of th...   

                                               STARS          VOTES  \
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21062.000000   
1  \n            \n    Stars:\nChris Wood, \nSara...   17870.000000   
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885805.000000   
3  \n            \n    Stars:\nJustin Roiland, \n...  414849.000000   
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...   15124.062722   

      RUNTIME     GENRE1      GENRE2                  GENRE3  \
0  121.000000     Action      Horror    Thriller               
1   25.000000  Animation      Action   Adventure               
2   44.000000      Drama      Horror    Thriller               
3   23.000000  Animation   Adventure      Comedy               
4   68.688539     Action       Crime      Horror               

                Director                   STARS1 STARS2  
0        Peter Thorwarth        Peri Baumeister,      ..  
1          not specified                 not sure     ..  
2          not specified                 not sure     ..  
3          not specified                 not sure     ..  
4  Matthias Schweighöfer  Matthias Schweighöfer,      ..

In [42]:
df.isnull().sum()

MOVIES      0
YEAR        0
RATING      0
ONE-LINE    0
STARS       0
VOTES       0
RUNTIME     0
GENRE1      0
GENRE2      0
GENRE3      0
Director    0
STARS1      0
STARS2      0
dtype: int64

In [43]:
del df['STARS']


In [44]:
df.head()

MOVIES         YEAR    RATING  \
0                        Blood Red Sky       (2021)  6.100000   
1  Masters of the Universe: Revelation     (2021– )  5.000000   
2                     The Walking Dead  (2010–2022)  8.200000   
3                       Rick and Morty     (2013– )  9.200000   
4                      Army of Thieves       (2021)  6.921176   

                                            ONE-LINE          VOTES  \
0  A woman with a mysterious illness is forced in...   21062.000000   
1  The war for Eternia begins again in what may b...   17870.000000   
2  Sheriff Deputy Rick Grimes wakes up from a com...  885805.000000   
3  An animated series that follows the exploits o...  414849.000000   
4  A prequel, set before the events of Army of th...   15124.062722   

      RUNTIME     GENRE1      GENRE2                  GENRE3  \
0  121.000000     Action      Horror    Thriller               
1   25.000000  Animation      Action   Adventure               
2   44.000000      Drama      Horror    Thriller               
3   23.000000  Animation   Adventure      Comedy               
4   68.688539     Action       Crime      Horror               

                Director                   STARS1 STARS2  
0        Peter Thorwarth        Peri Baumeister,      ..  
1          not specified                 not sure     ..  
2          not specified                 not sure     ..  
3          not specified                 not sure     ..  
4  Matthias Schweighöfer  Matthias Schweighöfer,      ..

In [45]:
df.YEAR=df.YEAR.str.split(pat='(').str[1]

In [46]:
df.YEAR

0            2021)
1          2021– )
2       2010–2022)
3          2013– )
4            2021)
           ...    
9993       2022– )
9995       2021– )
9996       2022– )
9997       2021– )
9998       2021– )
Name: YEAR, Length: 8989, dtype: object

In [47]:
df.YEAR=df.YEAR.str.split(pat=')').str[0]

In [48]:
df.YEAR

0            2021
1          2021– 
2       2010–2022
3          2013– 
4            2021
          ...    
9993       2022– 
9995       2021– 
9996       2022– 
9997       2021– 
9998       2021– 
Name: YEAR, Length: 8989, dtype: object

In [49]:
df.head()

MOVIES       YEAR    RATING  \
0                        Blood Red Sky       2021  6.100000   
1  Masters of the Universe: Revelation     2021–   5.000000   
2                     The Walking Dead  2010–2022  8.200000   
3                       Rick and Morty     2013–   9.200000   
4                      Army of Thieves       2021  6.921176   

                                            ONE-LINE          VOTES  \
0  A woman with a mysterious illness is forced in...   21062.000000   
1  The war for Eternia begins again in what may b...   17870.000000   
2  Sheriff Deputy Rick Grimes wakes up from a com...  885805.000000   
3  An animated series that follows the exploits o...  414849.000000   
4  A prequel, set before the events of Army of th...   15124.062722   

      RUNTIME     GENRE1      GENRE2                  GENRE3  \
0  121.000000     Action      Horror    Thriller               
1   25.000000  Animation      Action   Adventure               
2   44.000000      Drama      Horror    Thriller               
3   23.000000  Animation   Adventure      Comedy               
4   68.688539     Action       Crime      Horror               

                Director                   STARS1 STARS2  
0        Peter Thorwarth        Peri Baumeister,      ..  
1          not specified                 not sure     ..  
2          not specified                 not sure     ..  
3          not specified                 not sure     ..  
4  Matthias Schweighöfer  Matthias Schweighöfer,      ..

# if we want here we can concatenat the data but here no need

In [50]:
pd.concat([df['GENRE1'], df['GENRE2']], 
                  axis = 1)

GENRE1                 GENRE2
0          Action                 Horror
1       Animation                 Action
2           Drama                 Horror
3       Animation              Adventure
4          Action                  Crime
...           ...                    ...
9993        Drama   Thriller            
9995    Animation                 Action
9996  Documentary      Sport            
9997    Adventure                  Drama
9998    Adventure                  Drama

[8989 rows x 2 columns]

In [51]:
df.head()

MOVIES       YEAR    RATING  \
0                        Blood Red Sky       2021  6.100000   
1  Masters of the Universe: Revelation     2021–   5.000000   
2                     The Walking Dead  2010–2022  8.200000   
3                       Rick and Morty     2013–   9.200000   
4                      Army of Thieves       2021  6.921176   

                                            ONE-LINE          VOTES  \
0  A woman with a mysterious illness is forced in...   21062.000000   
1  The war for Eternia begins again in what may b...   17870.000000   
2  Sheriff Deputy Rick Grimes wakes up from a com...  885805.000000   
3  An animated series that follows the exploits o...  414849.000000   
4  A prequel, set before the events of Army of th...   15124.062722   

      RUNTIME     GENRE1      GENRE2                  GENRE3  \
0  121.000000     Action      Horror    Thriller               
1   25.000000  Animation      Action   Adventure               
2   44.000000      Drama      Horror    Thriller               
3   23.000000  Animation   Adventure      Comedy               
4   68.688539     Action       Crime      Horror               

                Director                   STARS1 STARS2  
0        Peter Thorwarth        Peri Baumeister,      ..  
1          not specified                 not sure     ..  
2          not specified                 not sure     ..  
3          not specified                 not sure     ..  
4  Matthias Schweighöfer  Matthias Schweighöfer,      ..

In [52]:
df.to_csv("cleanes_dataset.csv")